tempesta_fw/sock_srv.c

```diff
@@ -1018,6 +1018,14 @@ tfw_cfgop_setup_srv_group(void)
 		list_del(&srv->list);
 		tfw_sg_add(tfw_cfg_sg, srv);
 	}
+
+	/*
+	 * Set up an APM data for each server in the group. This must
+	 * be done before the scheduler data is set into the group.
+	 */	
+	if ((ret = tfw_for_each_srv(tfw_cfg_sg, tfw_apm_add_srv)) != 0)
+		return ret;
```

Comment:
Well, the server configuration handler is called before `tfw_sock_srv_start()` (see `tfw_cfg_start_mods()`). Since we want to start timers when APM data structures are constructed, there is no sense to call `tfw_apm_add_srv()` here.